src/fragments/lib/graphqlapi/js/optimistic-ui.mdx

```diff
@@ -0,0 +1,539 @@
+                deleteMutation.mutate({
+                  id: realEstateProperty.id,
+                })
+              }
+            >
+              Delete
+            </button>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  // NOTE: For test / sample cleanup purposes only (not for docs example)
+  async function deleteAll() {
+    //region: delete realEstateProperties:
+    const response = await API.graphql<
+      GraphQLQuery<ListRealEstatePropertiesQuery>
+    >({
+      query: queries.listRealEstateProperties,
+    });
+
+    console.log(
+      "RealEstateProperties to delete",
+      response?.data?.listRealEstateProperties?.items
+    );
+
+    await response?.data?.listRealEstateProperties?.items.forEach(
+      async (realEstateProperty) => {
+        if (!realEstateProperty?.id) return;
+
+        const realEstatePropertyDetails: DeleteRealEstatePropertyInput = {
+          id: realEstateProperty?.id,
+        };
+
+        const deletedRealEstateProperty = await API.graphql<
+          GraphQLQuery<DeleteRealEstatePropertyMutation>
+        >({
+          query: mutations.deleteRealEstateProperty,
+          variables: { input: realEstatePropertyDetails },
+        });
+
+        console.log("RealEstateProperty deleted: ", deletedRealEstateProperty);
+      }
+    );
+    //endregion
+
+    //region verify all deletes were successful:
+    const secondResponse = await API.graphql<
+      GraphQLQuery<ListRealEstatePropertiesQuery>
+    >({
+      query: queries.listRealEstateProperties,
+    });
+    console.log(
+      "RealEstateProperties should be empty:",
+      secondResponse?.data?.listRealEstateProperties?.items
+    );
+
+    if (secondResponse?.data?.listRealEstateProperties?.items?.length === 0) {
+      console.log("All deletes successful!");
+    }
+    //endregion
+  }
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        {isErrorQuery && <div>{"Problem loading Real Estate Properties"}</div>}
+        {isLoading && <div>{"Loading Real Estate Properties..."}</div>}
+        <ul>
+          {/* TODO: make this work with isSuccess */}
+          {isSuccess &&
+            realEstateProperties?.map((realEstateProperty) => {
+              if (!realEstateProperty) return null;
+              return (
+                <li key={realEstateProperty.id}>
+                  {realEstateProperty.name}
+                  <button
+                    onClick={() =>
+                      setCurrentRealEstatePropertyId(realEstateProperty.id)
+                    }
+                  >
+                    Detail View
+                  </button>
+                </li>
+              );
+            })}
+        </ul>
+        <button
+          onClick={() => {
+            createMutation.mutate({
+              name: `New Home ${Date.now()}`,
+              address: `${
+                alphabet[Math.floor(Math.random() * alphabet.length)]
+              } St`,
```

Comment:
Nit: One way to avoid this `alphabet` variable or is to use `window.prompt("...")`. THen customers can provide their own values without us building an entire form interface

Reply:
I'm trying to keep the UI interface as minimal as possible to highlight the API + TanStack integration, and not confuse readers with anything that's unnecessary. I'll perhaps replace this by updating the number in the address instead, but let me know if you disagree. Also, isolating this to a button click (i.e. excluding user input) allows us to demonstrate optimistic UI for rapid consecutive updates, which I think is useful.